### PR TITLE
Change to europe mode (2667)

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR adjust the key download to Europe mode.

Story: [2667](https://jira.itc.sap.com/browse/EXPOSUREAPP-2667)

## Checklist

* [x] This PR does not change text that hasn't been signed off with the RKI. Have a look at the pinned issue [440](https://github.com/corona-warn-app/cwa-app-ios/issues)
* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) - Pull Requests that are not linked to an issue with you as an assignee will be closed.
* [ ] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

### CODE REVIEW

**Review by marcussc — APPROVED** [review]

File: COMPILATION_CONDITIONS.md [hunk 1]
```diff
@@ -8,5 +8,4 @@ Our code uses several (custom) [ compilation conditions](https://help.apple.com/
...
 | `COMMUNITY` | `true` is the *Community* build configuration/schema has been selected. The *Community* configuration enables other developers to run the app in Simulator. |
 | `USE_DEV_PK_FOR_SIG_VERIFICATION` | `true` if the app should use the dev-public keys for the signature verfification. This is only set for testing and during development. In case this is set mistakenly to `true` the app simply stops working. |
-| `DISABLE_CERTIFICATE_PINNING` | `true` if certificate pinning should be disabled. This is **NEVER EVER** set for builds that will end up in the App Store. We only use this for debugging purposes. |
-| `EUROPEMODE` | `true` if europe mode should be enabled. If the exposure detection runs in europe mode, only key packages for country "EUR" are downloaded. Countries from the supported countries list are ignored. |
+| `DISABLE_CERTIFICATE_PINNING` | `true` if certificate pinning should be disabled. This is **NEVER EVER** set for builds that will end up in the App Store. We only use this for debugging purposes. |
```
Comment (whiskey) [inline COMPILATION_CONDITIONS.md:11]: Spoiler Alert: this whole document might change in future versions ;)

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift [hunk 2]
```diff
@@ -27,6 +27,7 @@ final class ExposureDetection {
 	private var completion: Completion?
 	private var progress: Progress?
 	private var countryKeypackageDownloader: CountryKeypackageDownloading
+	private let country = "EUR"
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift:30]: Should we leave a short(!) comment why this is this fixed "EUR" and not something two-letter coded?

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift [hunk 3]
```diff
@@ -47,70 +48,22 @@ final class ExposureDetection {
 		progress?.cancel()
 	}
 
-	private func getSupportedCountries(completion: @escaping ([Country]) -> Void) {
-		delegate?.exposureDetection(supportedCountries: { [weak self] result in
-			guard let self = self else { return }
-
+	private func downloadKeyPackages(completion: @escaping () -> Void) {
+		self.countryKeypackageDownloader.downloadKeypackages(for: country) { result in
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift:52]: I like `self` (not 'myself'), but is this needed here?

```suggestion
		countryKeypackageDownloader.downloadKeypackages(for: country) { result in
```

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift [hunk 4]
```diff
@@ -47,70 +48,22 @@ final class ExposureDetection {
...
-				dispatchGroup.leave()
-			}
-		}
-
-		dispatchGroup.notify(queue: .main) {
-			if let error = errors.first {
-				self.endPrematurely(reason: error)
-			} else {
+			case .failure(let didEndPrematurelyReason):
+				self.endPrematurely(reason: didEndPrematurelyReason)
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift:55]: like above:

```suggestion
				endPrematurely(reason: didEndPrematurelyReason)
```

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift [hunk 5]
```diff
@@ -149,44 +102,18 @@ final class ExposureDetection {
 	func start(completion: @escaping Completion) {
 		self.completion = completion
 
-		#if EUROPEMODE
-
 		activityState = .downloading
 
-		let countryIDs = ["EUR"]
-		self.downloadKeyPackages(for: countryIDs) { [weak self] in
+		self.downloadKeyPackages { [weak self] in
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift:107]: self…!?

**Review by whiskey — CHANGES_REQUESTED** [review]
Lots of removals – great!

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift:52]: No... good eyes! It's a leftover. The code was in a closure before. I'll clean this up.

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift (hunk 4 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift:55]: This one is inside of a closure. But thanks for pointing this our. I have made self weak.

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift:30]: Added a comment and linked to the corresponding discussion.

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift (hunk 5 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift:107]: Removed leftover from closure code.

File: COMPILATION_CONDITIONS.md (hunk 1 above)
Comment (ArturFriesen) [reply COMPILATION_CONDITIONS.md:11]: 🍿

**Review by whiskey — APPROVED** [review]

**Pull request merged**